austation/code/game/materialization.dm

```diff
@@ -0,0 +1,297 @@
+	if(blacklist[Opath])
+		visible_message("<span class='warning'>Entity too vague or dangerous to summon.</span>")
+		return
+	var/obj/item/shiritori_ball/ball = new(loc)
+	ball.prime_spawn(src, Opath, current_player)
+	current_player.put_in_hands(ball)
+	spent_objs[phrase] = Opath
+	current_letter = lowertext(phrase[length(phrase)])
+	playsound(src, 'sound/magic/clockwork/invoke_general.ogg', 100, TRUE)
+	switch_player()
+
+/obj/structure/table/mat_shiritori/proc/add_player(mob/living/carbon/human/H)
+	if(!istype(H))
+		return
+	if((H in players) || (H in knockouts))
+		return
+	players += H
+	to_chat(H, "<span class='notice'>Welcome to <b>Materialization Shiritori!</b>\n \
+				Similar to traditional Shiritori, but any valid word submitted by the player will materialize infront of them, as long as it exists in this world.</span>\n \
+				<span class='info'>The last player standing is declared as the winner, players can be eliminated in 3 different ways:\n \
+				<b>1:</b> Failing to name a valid entity within 30 seconds of the last player's turn.\n \
+				<b>2:</b> Attempting to summon an entity that has already been mentioned.\n \
+				<b>3:</b> Death.\n \
+				You cannot harm your opponents through <b>direct</b> means. All summoned entities will be destroyed once the game has concluded.</span>")
+	ADD_TRAIT(H, TRAIT_PACIFISM, "shiritori")
+	RegisterSignal(H, COMSIG_MOB_SAY, .proc/player_say)
+
+/obj/structure/table/mat_shiritori/proc/remove_player(mob/living/carbon/human/H)
+	players -= H
+	knockouts += H
+	if(H == current_player)
+		switch_player(length(players) > 2)
+	if(!QDELETED(H)) // in case they got gibbed or some other weird thing happened
+		REMOVE_TRAIT(H, TRAIT_PACIFISM, "shiritori")
+		UnregisterSignal(H, COMSIG_MOB_SAY)
+	if(length(players) == 1) // squid game
+		visible_message("<span class='boldannounce'>[current_player] has won the game!</span>")
+		playsound(src, 'sound/effects/gong.ogg', 100, 0, 5)
+		playsound(src, 'sound/effects/bamf.ogg', 100, 0, 5)
+		end_game()
+
+
+/obj/structure/table/mat_shiritori/proc/switch_player(alert = TRUE)
+	var/index = players.Find(current_player)
+	if(length(players) == index)
+		index = 1 // move back to the bottom of the list
+	else
+		index++
+	time_left = initial(time_left)
+	current_player = players[index]
+	if(QDELETED(current_player) || current_player.stat == DEAD)
+		remove_player(current_player)
+		return
+	if(alert)
+		if(turns)
+			to_chat(current_player, "<span class='notice'>It is now your turn.</span>")
+		else
+			say("[current_player] is starting.")
+			to_chat(current_player, "<span class='notice'>You're going first, you get to pick the first word.</span>")
+	turns++
+	playsound(src, 'sound/arcade/hit.ogg', 100, TRUE)
+
+/obj/structure/table/mat_shiritori/process(delta_time)
+	if(!active)
+		return PROCESS_KILL
+	for(var/mob/living/L as() in players)
+		if(QDELETED(L) || L.stat == DEAD)
+			remove_player(L)
+			continue
+	time_left -= delta_time SECONDS
```

Comment:
the SECONDS define converts seconds to deciseconds, not the other way around. You're actually going to be showing centiseconds here using it.
```suggestion
	time_left -= delta_time / 10
```